pkg/costmodel/metrics.go

```diff
@@ -511,6 +512,18 @@ func StartCostModelMetricRecording(a *Accesses) bool {
 				nodeSeen[labelKey] = true
 			}
 
+			loadBalancers, err := a.Model.GetLBCost(a.Cloud)
+			for lbKey, lb := range loadBalancers {
+				// TODO: parse (if necessary) and calculate cost associated with loadBalancer based on dynamic cloud prices fetched into each lb struct on GetLBCost() call
+				keyParts := getLabelStringsFromKey(lbKey)
+				namespace := keyParts[0]
+				serviceName := keyParts[1]
+				a.LBCostRecorder.WithLabelValues(namespace, serviceName).Set(lb.Cost)
+
+				labelKey := getKeyFromLabelStrings(namespace, serviceName)
+				loadBalancerSeen[labelKey] = true
+			}
+
```

Comment:
I haven't made any changes to `ComputeCostData()`. Do I need to?

It looks to me that `metrics.go:StartCostModelMetricRecording()` calls `getNodeCost()` to fetch cloud provider data, but actually does the total cost calculation in `StartCostModelMetricRecording()` (`metrics.go:500`). Is this the pattern we want to repeat going forward?

Reply:
ComputeCostData is building the allocation model, not the assets. Since the goal of this PR is to build up LoadBalancer assets in prometheus and not necessarily to perfectly allocate LB costs, this seems fine.